server/fleet/microsoft_mdm.go

```diff
@@ -1417,7 +1417,12 @@ func (cmd *SyncMLCmd) GetTargetData() string {
 	}
 
 	if cmd.Items[0].Data != nil {
-		return cmd.Items[0].Data.Content
+		content := cmd.Items[0].Data.Content
+		content = strings.TrimSpace(content)
+		content = strings.TrimPrefix(content, "<![CDATA[")
+		content = strings.TrimSuffix(content, "]]>")
+
+		return content
```

Comment:
Currently this is fine AFAICT since `GetTargetData` is only ever called in the profile verification process, but I think this could become error-prone if used in a different context now that it does not return the raw target data. I'd suggest renaming the method to something like `GetNormalizedTargetDataForVerification` ? Something that makes it clear that the data gets modified and is not the raw representation of the profile.

Reply:
Good callout, and a good new name which I think is clear from reading - I will make that adjustment.

Reply:
Actually will add a new method as we also use the `GetTargetData´ for the incoming SyncML commands, which should have this cleansed anyways, but for clarity sake making it a new method that is only called in the ExpectedHostPRofile verification part.